Simplify build process

bennofs opened the pull request:
This PR removes all dependencies on bower, npm and ruby. 
Not sure how to test this, `make migrate` fails with an unique index constraint violation on my system.

Feliix42 commented:
Huh? Can you give me the exact error and the involved versions of python, its modules and sqlite? 😅

bennofs commented:
I fixed a few issues with the original PR, it should work now (`make migrate` still fails, but I can load the site without successful migration).

```
  ...
  Applying user.0002_userinformation_accepted_privacy_policy... OK
  Applying user.0003_auto_20191025_1734... OK
  Applying user.0004_auto_20191026_0930... OK
env/bin/python manage.py loaddata courses
Traceback (most recent call last):
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.IntegrityError: UNIQUE constraint failed: django_content_type.app_label, django_content_type.model

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/code/course-management/env/lib/python3.7/site-packages/modeltranslation/management/commands/loaddata.py", line 61, in handle
    return super(Command, self).handle(*fixture_labels, **options)
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/management/commands/loaddata.py", line 69, in handle
    self.loaddata(fixture_labels)
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/management/commands/loaddata.py", line 109, in loaddata
    self.load_label(fixture_label)
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/management/commands/loaddata.py", line 175, in load_label
    obj.save(using=self.using)
  File "/code/course-management/env/lib/python3.7/site-packages/django/core/serializers/base.py", line 205, in save
    models.Model.save_base(self.object, using=using, raw=True, **kwargs)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/models/base.py", line 838, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/models/base.py", line 905, in _save_table
    forced_update)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/models/base.py", line 955, in _do_update
    return filtered._update(values) > 0
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/models/query.py", line 667, in _update
    return query.get_compiler(self.db).execute_sql(CURSOR)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1204, in execute_sql
    cursor = super(SQLUpdateCompiler, self).execute_sql(result_type)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 899, in execute_sql
    raise original_exception
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 889, in execute_sql
    cursor.execute(sql, params)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/code/course-management/env/lib/python3.7/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/code/course-management/env/lib/python3.7/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: Problem installing fixture '/code/course-management/course/fixtures/courses.yaml': Could not load contenttypes.ContentType(pk=11): UNIQUE constraint failed: django_content_type.app_label, django_content_type.model
make: *** [Makefile:42: migrate] Error 1
```

bennofs commented:
The issue with the migrations seems to be that the fixture contains data that is already added during the initial migration. That's why loaddata fails, since it tries to re-add data with an UNIQUE constraint.
